Automattic/simplenote-android
# Issue/844 add recent suggestions

### Fix
Add recent search queries to the search input field to close #844.  The recent search list is ordered from most recent at the top to least recent at the bottom.  Each submitted search is saved as a recent search with a maximum of five recent searches.  When a search is submitted after five recent searches have been saved, the least recent search is removed.  When a recent search is tapped, it becomes the most recent search.  Each recent search can be deleted with the ***X*** (i.e. ***Delete item***) button to the right of the search item row.  Deleting a recent search can be undone by tapping the ***Undo*** action in the ***Deleted recent search item*** snackbar.  See the screenshots and animation below for illustration.

![844_add_recent_suggestions](https://user-images.githubusercontent.com/3827611/68254426-c4745f80-ffe7-11e9-99a9-1e2e59387cb3.png)

<a href="https://user-images.githubusercontent.com/3827611/68254204-34361a80-ffe7-11e9-9abb-a39732934d6f.gif"><img src="https://user-images.githubusercontent.com/3827611/68254204-34361a80-ffe7-11e9-9abb-a39732934d6f.gif" width="320"></a>

### Test
1. Tap ***Search*** action in top app bar.
2. Notice no recent searches are shown.
3. Enter text into search field.
4. Notice suggestions appear based on entered text.
5. Tap any suggestion.
6. Notice suggestion text is inserted into search field.
7. Notice note list is shown based on inserted text.
8. Tap ***X*** button in search field.
9. Notice previously searched item is shown in recent searches.
10. Enter text into search field.
11. Tap magnifying glass button in keyboard.
12. Notice note list is shown based on input text.
13. Tap ***X*** button in search field.
14. Notice search from Step 10 is shown in recent searches followed by search from Step 5.
15. Tap recent search from Step 5.
16. Notice recent search text is inserted into search field.
17. Notice note list is shown based on inserted text.
18. Tap ***X*** button in search field.
19. Notice recent search from Step 5 is shown in recent searches follow by search from Step 10.
20. Long-press ***X*** button to right of any recent search.
21. Notice ***Delete item*** message is shown.
22. Tap ***X*** button to right of any recent search.
23. Notice recent search item from Step 22 is removed from recent searches.
24. Notice ***Deleted recent search item*** snackbar.
25. Tap ***Undo*** action in snackbar.
26. Notice recent search item from Step 22 is added to recent searches in previous position.

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release
`RELEASE-NOTES.txt` was updated in 7f2f9c7 with:
> Added search history and suggestions
